### PR TITLE
dev: download caddy binaries

### PR DESCRIPTION
**Pull request by keegancsmith**
This uses the same pattern we use for docsite and jaeger. This avoids including
all the dependencies of caddy as well as avoiding the need to build it.

### CODE REVIEW

**Comment by keegancsmith** [comment]
To get an idea of the number of deps this adds, see the diff to [go.sum](https://github.com/sourcegraph/sourcegraph/commit/5173cbc0a6686fb92007ad1e833a806a314f9865#diff-f949e2d81c8076ebbf8af38fcbb72c1f)

File: dev/caddy.sh [hunk 1]
```diff
@@ -1,14 +1,33 @@
...
-export GOBIN="$PWD/.bin"
-export PATH=$GOBIN:$PATH
-export GO111MODULE=on
+pushd "$(dirname "${BASH_SOURCE[0]}")/.." > /dev/null
 
-CADDY_PATH="${GOBIN}/caddy"
+mkdir -p .bin
 
-go build -o "${CADDY_PATH}" "github.com/caddyserver/caddy/v2/cmd/caddy"
+beta=20
```
Comment (unknwon) [inline dev/caddy.sh]: Caddy 2 Twitter states this is the last planned beta, so maybe not worthing having this, a single `version` string could be sufficient?

Comment (keegancsmith) [reply dev/caddy.sh]: I'll update after the official release rather. If you can get to 20 beta releases, 21 is not out of the question :P

Comment (unknwon) [reply dev/caddy.sh]: LOL, I want to wait until this merged before pulling the latest `master`.

**Pull request merged**